core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/workflow/WorkflowVersionResource.scala

```diff
@@ -343,4 +347,55 @@ class WorkflowVersionResource {
       res
     }
   }
+
+  @POST
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  @Path("/clone/{vid}")
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  def cloneVersion(
+      @PathParam("vid") vid: UInteger,
+      @Auth sessionUser: SessionUser,
+      requestBody: java.util.Map[String, Int]
+  ): UInteger = {
+    val offset = requestBody.get("offset")
+
+    // Fetch the workflow ID (`wid`) associated with the specified version (`vid`)
+    val versionRecord = Option(
+      context
+        .select(WORKFLOW_VERSION.WID)
+        .from(WORKFLOW_VERSION)
+        .where(WORKFLOW_VERSION.VID.eq(vid))
+        .fetchOne()
+    ).getOrElse {
+      throw new NotFoundException(s"Version ID $vid not found.")
+    }
+    val wid = versionRecord.get(WORKFLOW_VERSION.WID)
+    // Use retrieveWorkflowVersion to get the specified version of the workflow
+    val workflowVersion = retrieveWorkflowVersion(wid, vid, sessionUser)
+    // Generate a new name for the cloned workflow
+    val newWorkflowName = s"${workflowVersion.getName}_v${offset}_clone"
```

Comment:
Let's use the name *_copy instead of *_clone to match with the naming convention when user copies a workflow.